src/host_beta/messageRouter.ts

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+    encodeMessageForChannel,
+    FrameToolsEvent,
+    IOpenEditorData,
+    parseMessageFromChannel,
+    TelemetryData,
+    WebSocketEvent,
+    WebviewEvent,
+} from '../common/webviewEvents';
+
+declare const acquireVsCodeApi: () => {postMessage(message: unknown, args?: any|undefined): void};
+const vscode = acquireVsCodeApi();
+
+/**
+ * Both the DevTools iframe and the Extension (which owns the ws connection to the browser) will
+ * post messages to the Webview window for communication between the two. This class routes message
+ * to the correct location based on the origin and type of message posted.
+ *
+ */
+export class MessageRouter {
+    private toolsFrameWindow: Window | undefined;
```

Comment:
This is to make TS compiler happy. We don't have a reference to toolsFrameWindow at construction, only after DOMContentLoaded, so we have to mark this as possibly undefined. To your other comment above, we should consider adding an error reporting mechanism to allow this to be shown to the user if this ever fails to properly load, but I'd like that to be done separately from this, as we have parity with the legacy system in this approach

Reply:
made a slight change to accept null/undefined to prevent the need for turning null results from the getElementById call to undefined.

I agree this is not ideal and we should add error handling if the dt-window fails to load in a separate PR.